src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -741,37 +739,29 @@ static CHIP_ERROR EncodeSigma1(MutableByteSpan & buf)
     {                                                                                                                              \
         MutableByteSpan buf(mem.Get(), bufferSize);                                                                                \
-        EXPECT_EQ(EncodeSigma1<params>(buf), CHIP_NO_ERROR);                                                                       \
+        EXPECT_EQ(EncodeSigma1Helper<params>(buf), CHIP_NO_ERROR);                                                                 \
                                                                                                                                    \
         TLV::ContiguousBufferTLVReader reader;                                                                                     \
         reader.Init(buf);                                                                                                          \
+        ParsedSigma1 parsedSigma1;                                                                                                 \
                                                                                                                                    \
-        ByteSpan initiatorRandom;                                                                                                  \
-        uint16_t initiatorSessionId;                                                                                               \
-        ByteSpan destinationId;                                                                                                    \
-        ByteSpan initiatorEphPubKey;                                                                                               \
-        bool resumptionRequested;                                                                                                  \
-        ByteSpan resumptionId;                                                                                                     \
-        ByteSpan initiatorResumeMIC;                                                                                               \
-        CASESession session;                                                                                                       \
-        EXPECT_EQ(session.ParseSigma1(reader, initiatorRandom, initiatorSessionId, destinationId, initiatorEphPubKey,              \
-                                      resumptionRequested, resumptionId, initiatorResumeMIC) == CHIP_NO_ERROR,                     \
-                  params::expectSuccess);                                                                                          \
-        if (params::expectSuccess)                                                                                                 \
+        EXPECT_EQ(ParseSigma1(reader, parsedSigma1) == CHIP_NO_ERROR, params::kExpectSuccess);                                     \
```

Comment:
So... The _old_ setup did each parse on a new CASESession instance.  This setup is not.  But ParseSigma1 has side-effects on the members of the CASESession (see comments above), which means you can get cross-talk between the testcases now.

Changes requested for this part.  Given how ParseSigma1 works, each call must be on a brand-new CASESession instance.